### ISSUE TITLE
Unexpected token for spread operator.

### ISSUE THREAD

**Issue by ayrton** [eslint/eslint#2532]
``` js
// app/assets/javascripts/currentUser.js
variable = {
  a: 'a',
  b: 'b'
}

export default {
  toJSON: function() {
    return {...variable};
  }
};
```

Currently throws an unexpected token while this is a valid syntax.

``` sh
$ eslint app/assets/javascripts/currentUser.js

app/assets/javascripts/webpack/currentUser.js
  8:13  error  Unexpected token ..

✖ 1 problem (1 error, 0 warnings)
```

I did this on a fresh `npm install eslint`.

**Comment by hzoo** [comment]
So this is https://github.com/sebmarkbage/ecmascript-rest-spread ES7 and ESLint isn't supporting it at the moment? Duplicate of #2352.

**Comment by ilyavolodin** [comment]
@hzoo is correct. We do not support object spread.

**Comment by 3den** [comment]
any update about this? I saw on the docs that i can add `spread: true` but it still gives me the same error. I am using eslint 2.8.0 and below is mine .eslintrc.yml

```yaml
extends: "eslint:recommended"

env:
  browser: true
  es6: true
  jasmine: true

parserOptions:
  ecmaVersion: 6
  ecmaFeatures:
    jsx: true
    spread: true
```

**Comment by ilyavolodin** [comment]
Option you are looking for is `experimentalObjectRestSpread`. http://eslint.org/docs/user-guide/configuring#specifying-parser-options

**Comment by ayrton** [comment]
Hey thanks all for replying, I'm working with quite a big codebase, so it will take some time to update the dependencies. Once I do I will come back and report back here!

**Comment by 3den** [comment]
Thanks for the repplies, I fixed this bug by instaling babel-eslint.

**Comment by Salakar** [comment]
@ilyavolodin 's suggestion works for me 👍 

```json
{
  "extends": "airbnb",
  "parserOptions":{
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  }
}
```

**Comment by stevematdavies** [comment]
None of these options are working for me, I'm still getting the error, though my eslintrc file relfects all the suggestions recommended here!

Using eslint 3.11.1

**Comment by Salakar** [comment]
@stevematdavies try setting the `ecmaVersion` property under `parserOptions` to `6`

**Example:**
``` json
{
  "extends": "airbnb",
  "parserOptions":{
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  }
}
```

Source: [eslint docs - parser options](http://eslint.org/docs/user-guide/configuring#specifying-parser-options)

**Comment by AndrewKovalenko** [comment]
my eslint (v3.13.1) configuration is:

```
 "eslintConfig": {
    "parser": "babel-eslint",
    "parserOptions": {
      "ecmaVersion": 6,
      "sourceType": "module",
      "ecmaFeatures": {
        "jsx": true,
        "experimentalObjectRestSpread": true 
      }
    }
```

and it handles fine code like: 

```
const a  ={
   ...props
};
```

but if I try to do any props spread in JSX: 
```
        <MenuImplementation ...props/>

**Comment by masotime** [comment]
@AndrewKovalenko did you mean
```
<MenuImplementation {...props} />
```

**Comment by nickhallph** [comment]
Tried all the above, still erroring out.. :(
